balloc/balloc.c

```diff
@@ -2413,6 +2413,7 @@ balloc_regular_allocator(struct balloc_allocation_context *bac)
 	for (;cr < 3 && bac->bac_status == M0_BALLOC_AC_CONTINUE; cr++) {
 		M0_LOG(M0_DEBUG, "cr=%d", cr);
 		bac->bac_criteria = cr;
+		bac->bac_found = 0;
```

Comment:
Is it necessary to reset other members?
e.g. bac->bac_best?

Reply:
yes , we can reset it to make sure it recalculate best extent

Reply:
I saw you dropped this modification.
But I still think we need to reset "bac_best". 
When lock is released, the "bac_best" is out-dated. It might be invalid.
It must not be used any more.